Assignment2/Assignment2/ViewController.swift

```diff
@@ -9,36 +9,64 @@
 import UIKit
 
 class ViewController: RootViewController {
-
     //    Задача 0. Вывести наибольшее из двух чисел
     func findGreatestNumber(numberA: Int, numberB: Int) {
-        print("Greatest number is 9")
+        
+        if numberA > numberB {
+            print("Greatest number is \(numberA)")
+        }
+        else {
+            print("Greatest number is \(numberB)")
+        }
     }
-
+    
     //    Задача 1. Вывести квадрат и куб введенного числа
     func findSquareAndQubeOf(number: Int) {
-        fatalError()
+        var square = number * number
+        var qube = number * number * number
+        print("square of \(number) = \(square)\nqube of \(number) = \(qube)")
     }
-
+    
     //    Задача 2. Вывести все числа до заданного и в обратном порядке до 0
     func numbersUpToNAndBackToZero(with number: Int) {
-        fatalError()
+        for a in 0..<number {
+            print("\(a)")
+        }
+        for a in (0...number).reversed() {
+            print("\(a)")
+        }
     }
-
+    
     //    Задача 3. Подсчитать общее число делителей числа и вывести их
     func findDivisors(of number: Int) {
-        fatalError()
+        var devisorCount = 0
+        for devisor in 1..<number{
+            if number % devisor == 0 {
+                print(devisor)
+                devisorCount += 1
+            }
+        }
+        print("number \(number) has \(devisorCount) divisors")
     }
-
+    
     //    Задача 4 Проверить, является ли заданное число совершенным
-    func isNumberPerfect(_ number: Int) {
-        fatalError()
+    func isNumberPerfect(number: Int) {
+        var deviderCount2 = 0
+        var deviderSum = 0
+        
+        for devider in 1..<number {
```

Comment:
You can use number divided by two

Reply:
I thought to check all possible dividers there